django-stubs/db/models/deletion.pyi

```diff
@@ -57,7 +57,7 @@ class RestrictedError(IntegrityError):
 
 class Collector:
     using: str
-    data: Any
+    data: dict[str, Any]
```

Comment:
Looking at Django code, looks like the key is model class, not string?

```python
        # Initially, {model: {instances}}, later values become lists.
        self.data = defaultdict(set)
```

```python
    def add(self, objs, source=None, nullable=False, reverse_dependency=False):
        ...
        model = objs[0].__class__
        instances = self.data[model]
```

Also, incomplete types are fine, but looks like value be more precisely typed as `Collection[Any]` (common base class between list and set).

```suggestion
    data: dict[type[Model], Collection[Any]]
```

Reply:
Thanks for the review! The keys are indeed the model classes not strings. I have made the necessary changes. Just wanted to ask whether `Collection[Any]` is just there for increased type precision or it serves some other purpose as well? I was a bit curious about this part but couldn't figure this out.

Reply:
Collection is just for improved type precision.

Reply:
Perhaps there's some nuance to this that I'm missing, but why would you want to end up typing the values as `Any`? Everything in the dictionary's values is _supposed_ to be a homogenous bundle of `Model`, non?

To summarise what I put in the original ticket, AFAIK the _exact_ type match ought to be something like`Dict[Type[Model], Set[Model] | List[Model]]`, which fair enough, you might wish to relax down to `Collection`, but what necessitates throwing the `Any` type in?

Reply:
Hey! Thanks for reviewing my changes. I appreciate your feedback. To address your concern, the reason why "Any" was used instead of specifying the exact type of the values in the dictionary is because some parts of the codebase may use different data structures that inherit from the "Model" class. However, I see your point that it may be better to specify the type as "Set[Model] | List[Model]" for clarity. I will do the necessary changes. Thanks for bringing that up!